fetch.js

```diff
@@ -217,6 +217,11 @@
 
   function Request(url, options) {
     options = options || {}
+
+    if (url && url.match(/^\w+?:\/\/\w+?:\w+?@/)) {
+      throw new TypeError('Credentials not allowed in request URL')
+    }
+
```

Comment:
Using a regular expression to detect a little-used, edge case makes me worry about false positives. We should rely on the native `URL` class to handle parsing, and add a link to the [polyfill](https://github.com/components/url) in the README.md file for old browsers.

```javascript
parsed = new URL(url)
if (parsed.username || parsed.password) {
  throw new TypeError(...)
}
```